manager/integration/tests/test_basic.py

```diff
@@ -4038,6 +4038,82 @@ def test_default_storage_class_syncup():  # NOQA
     8. Revert the modifications of the ConfigMaps. Then wait for the
        StorageClass sync-up.
     """
+    def edit_configmap_allow_vol_exp(allow_exp):
+        """
+        allow_exp : bool, set allowVolumeExpansion
+        """
+        config_map = core_api.read_namespaced_config_map(
+                                                "longhorn-storageclass",
+                                                "longhorn-system")
+        config_map_data = yaml.safe_load(config_map.data["storageclass.yaml"])
+        config_map_data["allowVolumeExpansion"] = allow_exp
+        config_map.data["storageclass.yaml"] = yaml.dump(config_map_data)
+        core_api.patch_namespaced_config_map("longhorn-storageclass",
+                                             "longhorn-system",
+                                             config_map)
+
+        for i in range(RETRY_COMMAND_COUNT):
+            try:
+                longhorn_storage_class = storage_api.read_storage_class(
+                                                     "longhorn")
+                assert longhorn_storage_class.allow_volume_expansion is allow_exp # NOQA
+                break
+            except Exception as e:
+                print(e)
+            finally:
+                time.sleep(RETRY_INTERVAL)
+        longhorn_storage_class = storage_api.read_storage_class("longhorn")
+        assert longhorn_storage_class.allow_volume_expansion is allow_exp
+
+    def finalizer():
+
+        edit_configmap_allow_vol_exp(True)
+
+    request.addfinalizer(finalizer)
+
+    # step 1
+    storage_api = common.get_storage_api_client()
+    config_map = core_api.read_namespaced_config_map("longhorn-storageclass",
+                                                     "longhorn-system")
+    config_map_data = yaml.safe_load(config_map.data["storageclass.yaml"])
+
+    # step 2
+    longhorn_storage_class = storage_api.read_storage_class("longhorn")
+    longhorn_storage_class.reclaim_policy = "Retain"
+
+    # step 3
+    try:
+        not_allow_modify_reclaim_policy = False
+        storage_api.patch_storage_class("longhorn", longhorn_storage_class)
+    except Exception as e:
+        print(e)
+        not_allow_modify_reclaim_policy = True
```

Comment:
This part would not take effect, right? 

Because the patch operation will still work (there is no reason the API server will reject this, and we even don't have validation admission for storageclass resource), however, the storage class will be just reverted later to the content of the `longhorn-storageclass` configmap by longhorn controller.

see: https://github.com/longhorn/longhorn-manager/blob/c3e9b1c8c7d1043134bfbf2c8ceb25c789935bc0/controller/kubernetes_configmap_controller.go#L163-L201

Reply:
Hi @innobead ,
Yes this part is not take effect.

When I use API  `storage_api.patch_storage_class("longhorn", longhorn_storage_class)` I will get 
```
kubernetes.client.rest.ApiException: (422)
E           Reason: Unprocessable Entity
E           HTTP response headers: HTTPHeaderDict({'Audit-Id': 'd0cb4f21-3e7c-4f90-b677-04816c7423ef', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'X-Kubernetes-Pf-Flowschema-Uid': '9f5a14a6-7287-4b6e-8d74-e90cd4f009f5', 'X-Kubernetes-Pf-Prioritylevel-Uid': '15a86490-9398-463a-85cc-db72981ccb11', 'Date': 'Thu, 18 Aug 2022 09:15:14 GMT', 'Content-Length': '438'})
E           HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"StorageClass.storage.k8s.io \"longhorn\" is invalid: reclaimPolicy: Forbidden: updates to reclaimPolicy are forbidden.","reason":"Invalid","details":{"name":"longhorn","group":"storage.k8s.io","kind":"StorageClass","causes":[{"reason":"FieldValueForbidden","message":"Forbidden: updates to reclaimPolicy are forbidden.","field":"reclaimPolicy"}]},"code":422}
```
If I manually run `kubectl edit storageclass longhorn` and change  `reclaimPolicy: Delete` to
`reclaimPolicy: Retain`, console will reply 
```
longhorn-test:/integration/tests # kubectl edit storageclass longhorn
error: storageclasses.storage.k8s.io "longhorn" is invalid
A copy of your changes has been stored to "/tmp/kubectl-edit-gz60e.yaml"
error: Edit cancelled, no valid changes were saved.
```
So I think except in code is OK, I will keep try if I am in wrong usage of k8s API. thank you

Reply:
Hi @innobead ,
After discuss with developers, I figured out that step 3 means to edit reclaim policy in storage class's annotation `longhorn.io/last-applied-configmap`, not directly edit value of key `reclaimPolicy`
I updated code for review, thank you.